waspc/packages/ts-inspect/src/exports.ts

```diff
@@ -0,0 +1,93 @@
+import ts from 'typescript';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import JSON5 from 'json5';
+import { z } from 'zod';
+
+export const ExportRequest = z.object({
+  tsconfig: z.string().optional(),
+  filenames: z.array(z.string())
+});
+
+export const ExportRequests = z.array(ExportRequest);
+
+export type ExportRequest = z.infer<typeof ExportRequest>;
+
+export type Export
+  = { type: 'default' } & Location
+  | { type: 'named', name: string } & Location
+
+export type Location = { location?: { line: number, column: number } }
+
+export async function getExportsOfFiles(request: ExportRequest): Promise<{ [file: string]: Export[] }> {
+  let compilerOptions: ts.CompilerOptions = {};
+
+  // If a tsconfig is given, load the configuration.
+  if (request.tsconfig) {
+    const configJson = JSON5.parse(await fs.readFile(request.tsconfig, 'utf8'));
+    const basePath = path.dirname(request.tsconfig)
+
+    const { options, errors } = ts.convertCompilerOptionsFromJson(
+      configJson.compilerOptions, basePath, request.tsconfig
+    );
+    if (errors && errors.length) {
+      throw errors;
+    }
+    compilerOptions = options;
+  }
+
+  const exportsMap: { [file: string]: Export[] } = {};
+
+  // Initialize the TS compiler.
+  const program = ts.createProgram(request.filenames, compilerOptions);
+  const checker = program.getTypeChecker();
+
+  // Loop through each given file and try to get its exports.
+  for (let filename of request.filenames) {
+    try {
+      exportsMap[filename] = getExportsForFile(program, checker, filename);
+    } catch (err) {
+      console.error(err);
+      exportsMap[filename] = [];
+      continue;
```

Comment:
@craigmc08 I think `continue` doesn't do anything here, the loop continues irregardless

Reply:
Deleted `continue`